### PR TITLE
Generate default record constructors.

### PR DESCRIPTION
**Pull request by andreyk0**
Record constructors that take only the required fields. Maybe we can include this in `0.1.0` ?

### CODE REVIEW

File: src/Printer/PureScript/NewType.purs [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 
 import Prelude
-import Data.Array (elem)
-import Data.Foreign.NullOrUndefined (NullOrUndefined(..), unNullOrUndefined)
-import Data.Maybe (Maybe(..), fromMaybe, maybe)
-import Data.String (Pattern(Pattern), Replacement(Replacement), drop, dropWhile, joinWith, replace, replaceAll, take, toUpper)
-import Data.StrMap (StrMap, isEmpty, filterKeys, toArrayWithKey)
 
 import Aws (MetadataElement(MetadataElement), ServiceShape(ServiceShape), ServiceShapeName(ServiceShapeName))
+import Data.Array (elem, partition)
```
Comment (plippe) [inline src/Printer/PureScript/NewType.purs]: Not sure if it is a right thing, I usually split imports based on their "origin". External ones first, my own after. Would that be good practice, or something useless ?

File: src/Printer/PureScript/NewType.purs [hunk 2]
```diff
@@ -87,20 +90,60 @@ recordMap (ServiceShapeName value) = "(StrMap.StrMap {{value}})"
...
+    then "" -- there's already a singleton constructor
+    else """
+-- | Constructs {{name}} from required parameters
+new{{name}} :: {{newTypeSignature}}
+new{{name}} {{arguments}} = {{name}} { {{fieldAssignments}} }
+
+-- | Constructs {{name}}'s fields from required parameters
+--   This may be useful if you need to immediately overwrite some of the optional values
+new{{name}}' :: {{fieldsSignature}}
+new{{name}}' {{arguments}} customize = ({{name}} <<< customize) { {{fieldAssignments}} }
```
Comment (plippe) [inline src/Printer/PureScript/NewType.purs]: Never saw this before. Is that a haskell convention type of thing ?

**Comment by plippe** [comment]
Looks great, I would say, merge away.

Once the CD finally works, we can change to 0.1.0. Don't really want to release anything that won't take care of itself.

File: src/Printer/PureScript/NewType.purs (hunk 1 above)
Comment (andreyk0) [reply src/Printer/PureScript/NewType.purs]: I've been using spacemacs with purescript IDE integration, it has some built-in behaviors for imports, so, it's whatever it did.

File: src/Printer/PureScript/NewType.purs (hunk 2 above)
Comment (andreyk0) [reply src/Printer/PureScript/NewType.purs]: Not really, something I've made up for now.
Typically one would use lenses but it's a bit more work to auto-generate those, so, I've added this for now.
More often than not there's a need to set at least one optional parameter, so, without some support one would have to create "wrapped" type, unwrap, modify record, re-wrap. Lenses solve that problem neatly but this allows for a slightly less awkward syntax with just the record APIs.

e.g. `_ { foo = 1 }` would be a function "from record to record" that can be given as an argument to customize input.

**Comment by andreyk0** [comment]
Cool, I'm going to merge for now. We can keep tweaking things as we use the API more.
It's all "version zero" anyway, we can change our minds about some things.

**Comment by andreyk0** [comment]
sorry, clicked wrong button, closed/reopened as a result.

**Pull request merged**